dlt/common/storages/fsspec_filesystem.py

```diff
@@ -157,29 +199,31 @@ class FileItemDict(DictStrAny):
     def __init__(
         self,
         mapping: FileItem,
-        credentials: Optional[Union[FileSystemCredentials, AbstractFileSystem]] = None,
+        fs_details: Optional[Union[AbstractFileSystem, FilesystemConfiguration, FileSystemCredentials]] = None,
```

Comment:
I have simply added `FilesystemConfiguration` as an additional type, and tidied up the naming.  Is that far enough for this PR?

Possible fast-follower to this PR:
1. Limit the types that can be passed to `FileItemDict` constructor. I get the feeling this is an important internal API. The following calls are currently in use:
- `AbstractFileSystem`, in tests/common/storages/test_fsspec_filesystem.py
- `FileSystemCredentials`, in tests/common/storages/utils.py
- `None`, in verified-sources sources/inbox/__init__.py
Could there be other usage in dlt example projects?